relay-server/src/actors/processor.rs

```diff
@@ -3075,11 +3150,60 @@ impl EnvelopeProcessorService {
         }
     }
 
+    #[cfg(not(feature = "processing"))]
+    fn handle_encode_metric_meta(&self, message: EncodeMetricMeta) {
+        let EncodeMetricMeta { scoping, meta } = message;
+
+        let upstream = self.inner.config.upstream_descriptor();
+        let dsn = crate::extractors::PartialDsn {
+            scheme: upstream.scheme(),
+            public_key: scoping.project_key,
+            host: upstream.host().to_owned(),
+            port: upstream.port(),
+            path: "".to_owned(),
+            project_id: Some(scoping.project_id),
+        };
+
+        let mut item = Item::new(ItemType::MetricMeta);
+        item.set_payload(ContentType::Json, serde_json::to_vec(&meta).unwrap());
+        let mut envelope = Envelope::from_request(None, RequestMeta::outbound(dsn));
+        envelope.add_item(item);
+        let envelope = ManagedEnvelope::standalone(envelope, Addr::dummy(), Addr::dummy());
```

Comment:
In a follow-up, replace `Addr::dummy` with the actual services. #2682 adds the boilerplate needed for this.